src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp

```diff
@@ -1002,8 +1007,8 @@ namespace OpenMS
                         mod = xl_db->getModification(mods[s], "", ResidueModification::C_TERM);
                       }
                     }
-                    acc = mod.getPSIMODAccession();
-                    name = mod.getId();
+                    acc = mod->getPSIMODAccession();
```

Comment:
```suggestion
                    // mod should never be null, but gcc complains (-Werror=maybe-uninitialized)
                    if (mod != nullptr) acc = mod->getPSIMODAccession();
```

for consistency's sake at least check both dereferences of `mod`?